src/ser.rs

```diff
@@ -411,10 +411,22 @@ where
     #[inline]
     fn serialize_tuple_struct(
         self,
-        _name: &'static str,
+        name: &'static str,
         len: usize,
     ) -> Result<&'a mut Serializer<W>> {
-        self.serialize_tuple(len)
+        use tagged::CBOR_NO_LENGTH_FIELD;
+        if name.as_ptr() == CBOR_NO_LENGTH_FIELD.as_ptr() && name == CBOR_NO_LENGTH_FIELD {
```

Comment:
What's the point of comparing the pointers?

Reply:
My primary goal was to avoid external users from getting inside the `if` statement. Secondary goal was speed. After having slept on it, the first seems like a stupid (and unobtainable) goal, while the second will probably not matter at all.